File: App/Resources/AlgorithmDetails/slopesort.bundle/slopesort.kt
fun sort(arr: Array<Int>) {
  val n = arr.size
  for (start in 1 until n) {
    var i = start
    var k = start - 1
    while (k >= 0) {
      if (arr[i] < arr[k]) {
        arr[i] = arr[k].also { arr[k] = arr[i] }
      }
      k--
      i--
    }
  }
}

fun main() {
  var array = arrayOf<Int>(0, 39, 21, 62, 91, 77, 14, 23,
    90, 69, 51, 81, 68, 83, 32, 56)
  sort(array)
  println("[%s]".format(array.joinToString(", ")))
}

File: App/Resources/AlgorithmDetails/slopesort.bundle/slopesort.py
def sort(arr):
  n = len(arr)
  for start in range(1, n):
    i = start
    k = start - 1
    while k >= 0:
      if arr[i] < arr[k]:
        arr[i], arr[k] = arr[k], arr[i]
      k -= 1
      i -= 1

if __name__ == "__main__":
  array = [0, 39, 21, 62, 91, 77, 14, 23,
    90, 69, 51, 81, 68, 83, 32, 56]
  sort(array)
  print(array)